ui/src/AppCharting/LineChartjs.svelte

```diff
@@ -0,0 +1,223 @@
+
+    const selectedSet = new Set(selectedTimestamps);
+    const bySize = chartType === BY_SIZE;
+
+    const xLabels = data.columns[0].slice(1);
+
+    const active = [...errorTicks.values()].filter(error => {
+      if (!error.length) return false;
+      const {scenario: es, key} = error[0];
+      return es === scenario && selectedSet.has(key);
+    });
+
+    for (const error_list of active) {
+      for (const error of error_list) {
+        const {key, scenario: _scenario, dateTime, size} = error;
+
+        const timestampAsClassName = classNameFromBenchmarkKey(key);
+
+        const trimmedSize = JSON.stringify(size);
+        const formattedDateTime = dateTime.replace('_', ' ');
+        let className = bySize ? timestampAsClassName : trimmedSize;
+        const label = bySize ? trimmedSize : formattedDateTime;
+        className = formatTime(className);
+        const index = xLabels.indexOf(label);
+        if (index && label && className) {
+          formattedErrors.push({index, label, className});
+        }
+      }
+    }
+  }
+
+  function onRendered(theKey: any): void {
+    if (theKey !== key) {
+      console.debug("There's a different ref in town, skipping", theKey, key);
+      return;
+    }
+    dispatch('rendered');
+  }
+
+  async function drawChart(data, axis): Promise<void> {
+    const xAxis = axis.x;
+    const yAxis = axis.y;
+    const xAxisLabel = xAxis.label.text;
+    const yAxisLabel = yAxis.label.text;
+    const yAxisType = yAxis.type === 'log' ? 'logarithmic' : yAxis.type;
+
+    if (errorTicks) formatErrors();
+    let datasets = [];
+    let xValues = [];
+
+    let colors = [
+      'olivegreen',
+      'darkorange',
+      'darkgreen',
+      'purple',
+      'darkblue',
+      'gold',
+      'magenta',
+      'blueviolet',
+      'darkred',
+      'sienna'
+    ];
+
+    if (data && data.names && data.columns.length) {
+      xValues = data.columns[0].slice(1);
+      for (let i = 1; i < data.columns.length; i++) {
+        let column = data.columns[i];
+        let label = column[0];
+
+        // format time for chart legend
+        label = formatTime(label);
+        let container = {
+          data: [],
+          label,
+          backgroundColor: colors[i],
+          borderColor: colors[i],
+          pointBackgroundColor: Array(column[0].length).fill(colors[i]),
+          pointRadius: 5
+        };
+
+        // set error point color
+        for (const error of formattedErrors) {
+          if (error.className === label) {
+            container.pointBackgroundColor[error.index] = 'red';
+          }
+        }
+
+        let dataSet = [];
+        let yValues = column.slice(1);
+        xValues.forEach((x, index) => {
+          let y = yValues[index];
+          dataSet.push({x, y});
+        });
+
+        container.data = dataSet;
+        datasets.push(container);
+      }
+    }
+
+    chartRef?.destroy();
+    const theKey = Date.now();
```

Comment:
Do you need both `theKey` and `key`?

Reply:
Thanks for reminding me of this, I was actually able to remove both of those variables and a few other lines of code